src/composables/useGetParticipants.js

```diff
@@ -0,0 +1,135 @@
+/**
+ * @copyright Copyright (c) 2024 Dorra Jaouad <dorra.jaoued1@gmail.com>
+ *
+ * @author Dorra Jaouad <dorra.jaoued1@gmail.com>
+ * @author Marco Ambrosini <marcoambrosini@icloud.com>
+ *
+ * @license AGPL-3.0-or-later
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+import debounce from 'debounce'
+import { ref, onMounted, nextTick, computed } from 'vue'
+
+import { subscribe, unsubscribe } from '@nextcloud/event-bus'
+
+import { useIsInCall } from './useIsInCall.js'
+import { useStore } from './useStore.js'
+import { CONVERSATION } from '../constants.js'
+import { EventBus } from '../services/EventBus.js'
+
+/**
+ * @param {boolean} isActive whether the participants tab is active
+ */
+export function useGetParticipants(isActive = true) {
+
+	// Encapsulation
+	const store = useStore()
+	const token = computed(() => store.getters.getToken())
+	const conversation = computed(() => store.getters.conversation(token.value))
+	const isInCall = useIsInCall()
+	const oneToOneConversation = computed(() => conversation.value.type === CONVERSATION.TYPE.ONE_TO_ONE)
+
+	// External use
+	const participantsInitialised = ref(false)
+	const fetchingParticipants = ref(false)
+	const pendingChanges = ref(false)
+	let debounceFastUpdateParticipants = ref(null)
+	let debounceSlowUpdateParticipants = ref(null)
```

Comment:
Having a `ref` variable defined by `let` is a flag of a possible issue.

`ref` is a reactive variable, object-container for a value. It has reactivity, but it works only while it is used as a const reference for a mutable object. So it must be a `const`, `ref` should never be re-defined.

But here these variables should not be `ref`s, because they don't store a reactive value, they are constant functions.

(See also next comment).

Reply:
Thank you for all explanations :)